packages/ui/src/server/controllers/vcs.ts

```diff
@@ -0,0 +1,320 @@
+import {AppRouteHandler, Request, Response} from '@gravity-ui/expresskit';
+import {VCSSettings} from '../../shared/ui-settings';
+import {AxiosError} from 'axios';
+import {AppConfig} from '@gravity-ui/nodekit';
+import {GithubRepository, GitlabRepository, VcsConfig} from '../../shared/vcs';
+import {sendError} from '../utils';
+import {VcsFactory} from '../components/vcs';
+
+const COOKIE_OPTIONS = {
+    httpOnly: true,
+    secure: true,
+    sameSite: 'strict' as const,
+    maxAge: 1000 * 60 * 60 * 24 * 365,
+};
+
+const ERROR_MESSAGE = {
+    VCS_CONFIG_NOT_FOUND: 'Vcs config is not found',
+    VCS_ID_REQUIRED: 'Vcs id is required',
+    VCS_NOT_SUPPORTED: 'This vcs id is not supported',
```

Comment:
As I understand `VCS_NOT_SUPPORTED` is used when a specific `vcsId` is not configured, so the message should contain something about configuration.